storybook/stories/Components.tsx

```diff
@@ -30,7 +32,17 @@ export default () =>
             : <div>loading...</div>
           )}
         </DAO.Data>
+        <MemberWrapper address="0xcb4e66eca663fdb61818d52a152601ca6afef74f">
+          <Member.Data>
+            {(data: MemberData | undefined) => (
+              data ?
+              <div>{data.reputation.toString()}</div>
```

Comment:
How does the user of this method know what the structure of the `data` object will be? One way would be to have a method at the `MemberWrapper` layer that takes the named keys and turns them into props. Would this affect the type mapping in a negative way (since you would have to supply a context)?

Reply:
I don't quite understand your proposed implementation, would like to look at the code with you and hear you explain it further.

I do agree though, the typing of the function's argument isn't well enforced or inferred. It defaults to being (in this case) MemberData, and intellisense will respect that, but the `| undefined` part needs to be known by the user. This has been something I've been wanting to abstract away through, so it might become a non issue.

Reply:
@gh1dra "named keys and turns them into props" you're a genius! lol, just stumbled upon this solution you were getting at.